components/kyma-environment-broker/internal/process/provisioning/event_hub_step.go

```diff
@@ -0,0 +1,208 @@
+	if err != nil {
+		// internal error, repeating doesn't solve the problem
+		errorMessage := fmt.Sprintf("Failed to create Azure EventHubs client: %v", err)
+		return p.operationManager.OperationFailed(operation, errorMessage)
+	}
+
+	// create Resource Group
+	groupName := pp.Parameters.Name
+	// TODO(nachtmaar): use different resource group name https://github.com/kyma-incubator/compass/issues/967
+	resourceGroup, err := namespaceClient.CreateResourceGroup(p.context, azureCfg, groupName)
+	if err != nil {
+		// retrying might solve the issue while communicating with azure, e.g. network problems etc
+		errorMessage := fmt.Sprintf("Failed to persist Azure Resource Group [%s] with error: %v", groupName, err)
+		return p.retryOperation(operation, errorMessage, time.Minute, time.Minute*30, log)
+	}
+	log.Printf("Persisted Azure Resource Group [%s]", groupName)
+
+	// create EventHubs Namespace
+	eventHubsNamespace := pp.Parameters.Name
+	eventHubNamespace, err := namespaceClient.CreateNamespace(p.context, azureCfg, groupName, eventHubsNamespace)
+	if err != nil {
+		// retrying might solve the issue while communicating with azure, e.g. network problems etc
+		errorMessage := fmt.Sprintf("Failed to persist Azure EventHubs Namespace [%s] with error: %v", eventHubsNamespace, err)
+		return p.retryOperation(operation, errorMessage, time.Minute, time.Minute*30, log)
+	}
+	log.Printf("Persisted Azure EventHubs Namespace [%s]", eventHubsNamespace)
+
+	// get EventHubs Namespace secret
+	accessKeys, err := namespaceClient.GetEventhubAccessKeys(p.context, *resourceGroup.Name, *eventHubNamespace.Name, authorizationRuleName)
+	if err != nil {
+		// retrying might solve the issue while communicating with azure, e.g. network problems etc
+		errorMessage := fmt.Sprintf("Unable to retrieve access keys to azure event-hub namespace: %v", err)
+		return p.retryOperation(operation, errorMessage, time.Minute, time.Minute*30, log)
+	}
+	kafkaEndpoint := extractEndpoint(accessKeys)
+	kafkaEndpoint = appendPort(kafkaEndpoint, kafkaPort)
+	kafkaPassword := *accessKeys.PrimaryConnectionString
+
+	// set installation overrides
+	operation.InputCreator.SetOverrides(componentNameKnativeEventing, getKnativeEventingOverrides())
+	operation.InputCreator.SetOverrides(componentNameKnativeEventingKafka, getKafkaChannelOverrides(kafkaEndpoint, k8sSecretNamespace, "$ConnectionString", kafkaPassword, kafkaProvider))
+
+	return operation, 0, nil
+}
+
+func (p *ProvisionAzureEventHubStep) retryOperationOnce(operation internal.ProvisioningOperation, wait time.Duration) (internal.ProvisioningOperation, time.Duration, error) {
+	return operation, wait, nil
+}
+
+// retryOperation retries an operation for at maxTime in retryInterval steps
+func (p *ProvisionAzureEventHubStep) retryOperation(operation internal.ProvisioningOperation, errorMessage string, retryInterval time.Duration, maxTime time.Duration, log logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
```

Comment:
@jasiu001 IMO those functions are quite generic and should be provided by the framework itself, WDYT?

Reply:
I agree it could be used in other places/steps. Maybe the good place for this kind of method is `/internal/process/operation.go`

Reply:
That sounds good! I think it is easier to read than just returning 3 arguments. I will place both methods to operation.go then.

Reply:
We should have a follow up PR then to make use of these methods.

E.g. [here for RetryOperation](https://github.com/kyma-incubator/compass/blob/b13d9c2a0a4b6a2368f8cd7edef0112a92eb51f5/components/kyma-environment-broker/internal/process/provisioning/resolve_creds.go#L78
) and [here for retryOnce](https://github.com/kyma-incubator/compass/blob/b13d9c2a0a4b6a2368f8cd7edef0112a92eb51f5/components/kyma-environment-broker/internal/process/provisioning/resolve_creds.go#L102)

Reply:
Done. I have the logger as method argument which does not look too nice. But since the logger comes the steps Run method and is not inside the steps structure for all steps which we have atm, it is the easiest solution for now.